ast_parser.py

```diff
@@ -188,17 +195,22 @@ def parse_param_list(self):
                 node = self.parse_parameter()
                 if node:
                     root.add_child(node)
-                if self.current_token()[1] == ')':
-                    root.add_child(ASTNode('SEPARATOR', ')'))
-                    self.advance()
-                    break
```

Comment:
Imagine a user input like: (a, b, c,) 
I believe that your current logic will treat it as a valid param_list

This might be an issue for {} and [] as well

Reply:
{} and() is working fine. I will adjust the logic for other user input thing